auto-cancel.md

```diff
@@ -0,0 +1,164 @@
+The *auto-cancel* strategy defines the behavior of pipelines when they enter the
+queue for execution.
+
+By default, all pipelines will be queued and executed on a first-come, first-served
+basis.
```

Comment:
This is still true with the auto-cancel feature too. The FIFO order is not broken. Auto-cancel only "speeds up" the FIFO processing, and cancels redundant items.

Reply:
The later sentence should be related more with `all pipelines will be queued` part than FIFO, but I agree it is not so clear. I will rework that part.